subsys/testsuite/ztest/include/ztest_test_new.h

```diff
@@ -0,0 +1,323 @@
+		.name = #SUITE_NAME,                                                               \
+		.setup = (setup_fn),                                                               \
+		.before = (before_fn),                                                             \
+		.after = (after_fn),                                                               \
+		.teardown = (teardown_fn),                                                         \
+		.predicate = PREDICATE,                                                            \
+	}
+
+/**
+ * Run the registered unit tests which return true from their pragma function.
+ *
+ * @param state The current state of the machine as it relates to the test executable.
+ * @return The number of tests that ran.
+ */
+int ztest_run_test_suites(const void *state);
+
+/**
+ * @brief Fails the test if any of the registered tests did not run.
+ *
+ * When registering test suites, a pragma function can be provided to determine WHEN the test should
+ * run. It is possible that a test suite could be registered but the pragma always prevents it from
+ * running. In cases where a test should make sure that ALL suites ran at least once, this function
+ * may be called at the end of test_main(). It will cause the test to fail if any suite was
+ * registered but never ran.
+ */
+void ztest_verify_all_test_suites_ran(void);
+
+/**
+ * @brief Run a test suite.
+ *
+ * Internal implementation. Do not call directly. This will run the full test suite along with some
+ * checks for fast failures and initialization.
+ *
+ * @param name The name of the suite to run.
+ * @return Negative value if the test suite never ran; otherwise, return the number of failures.
+ */
+int z_ztest_run_test_suite(const char *name);
+
+/**
+ * @defgroup ztest_test Ztest testing macros
+ * @ingroup ztest
+ *
+ * This module eases the testing process by providing helpful macros and other
+ * testing structures.
+ *
+ * @{
+ */
+
+/**
+ * @brief Fail the currently running test.
+ *
+ * This is the function called from failed assertions and the like. You
+ * probably don't need to call it yourself.
+ */
+void ztest_test_fail(void);
+
+/**
+ * @brief Pass the currently running test.
+ *
+ * Normally a test passes just by returning without an assertion failure.
+ * However, if the success case for your test involves a fatal fault,
+ * you can call this function from k_sys_fatal_error_handler to indicate that
+ * the test passed before aborting the thread.
+ */
+void ztest_test_pass(void);
+
+/**
+ * @brief Skip the current test.
+ *
+ */
+void ztest_test_skip(void);
+
+/**
+ * @brief Do nothing, successfully.
+ *
+ * Unit test / setup function / teardown function that does
+ * nothing, successfully. Can be used as a parameter to
+ * ztest_unit_test_setup_teardown().
+ */
+static inline void unit_test_noop(void)
+{
+}
+
+#define Z_ZTEST(suite, fn, t_options)                                                              \
```

Comment:
maybe it would be worth to merge those macros:
```
#define Z_TEST(suite, fn, t_options, use_fixture) \
static void suite##_##fn(COND_CODE_1(use_fixture, (struct fixture *data), (void)); 
...
static inline void suite##_##fn(COND_CODE_1(use_fixture,(struct suite##_fixture *this), (void *)))
```

Reply:
Done